### DISCUSSION TITLE
Arrows, on-screen text and straight lines

### DISCUSSION THREAD

**Discussion by smapelli** [geovens/gInk#143] [Ideas]
It would be great if the tool would allow

- drawing arrows
- writing text
- drawing straight lines when drawing with some key selected (e.g. the shift key)

These additional functionalities would probably make this tool the best free on-screen annotation tool for the Windows platform :)

**Comment by pubpub-zz** [comment]
Hi @smapelli
I made a fork that contains those capabilities and more. Have a look at https://github.com/pubpub-zz/ppInk

**Reply by smapelli** [reply]
@pubpub-zz, great job! I'll test it out immediately.

**Reply by smapelli** [reply]
@pubpub-zz it seems to me that your release is somewhat buggy. For example the palette even slipped a few times below the task bar in my tests. 

![image](https://user-images.githubusercontent.com/40613147/144303401-0a5380e2-7592-46ce-9cf8-6232a93a8e1a.png)

I was unable to understand how to type on-screen text :) How does it work? 
Also is it wanted for the palette to collapse in the position displayed on the screenshot (not at the very far right) ?

If I may say it, it seems to me that buttons are a bit too crowded in your implementation. I preferred somewhat more the original layout of buttons.

**Reply by pubpub-zz** [reply]
> @pubpub-zz it seems to me that your release is somewhat buggy. For example the palette even slipped a few times below the task bar in my tests.
> 
> ![image](https://user-images.githubusercontent.com/40613147/144303401-0a5380e2-7592-46ce-9cf8-6232a93a8e1a.png)
> 

Unless I'm wrong the example you've produced, you are in pointer mode and you have two topmost window (ppInk and the taskbar) who are in competition. I'm not sure gInk is doing better.

> I was unable to understand how to type on-screen text :) How does it work?
Select the tool
![image](https://user-images.githubusercontent.com/4083478/144311006-9408dc86-815e-4554-bd80-c5f21e12959f.png)
then click where you want to type in your text. a dialogbox will open:
![image](https://user-images.githubusercontent.com/4083478/144312424-d59845e5-a964-46eb-9972-dcb80c692c54.png)
then type in your text

 Also is it wanted for the palette to collapse in the position displayed on the screenshot (not at the very far right) ?
By default the toolbar collapse to right. and the undock button is available at the top right position. if you want to change its position move it by drag and drop
> 
> If I may say it, it seems to me that buttons are a bit too crowded in your implementation. I preferred somewhat more the original layout of buttons.

I've introduced lots of new functions and I had to find a solution to not spread so much, that I've proposed this layout. if you think it is too much, hide the buttons through the options dialogbox ; the keyboard shortcuts will still be available.

**Reply by smapelli** [reply]
For the on-screen text, I believe the icon you selected is not that intuitive. In other applications the icon displays a letter which is, I think, easier to interpret. As for the competing topmost windows, why not simply avoid overlapping them? Don't get me wrong, I understand your intent in overlapping those windows but you are not gaining that much space anyway in having them partially overlap. Have you tried setting the taskbar as the parent of your window? Not sure it helps in avoiding yours getting below the other.

**Reply by pubpub-zz** [reply]
Icons can be customized. In next release, the icon will be replaced with a "T" 😊 (https://github.com/pubpub-zz/ppInk/issues/139#issuecomment-913017938 )
about topmost,not sure something can be done (see https://social.msdn.microsoft.com/Forums/en-US/4ca39c4e-a84f-4532-943a-fae16a06ab12/taskbar-cannot-set-as-top-most-window-using-bringwindowtotop-api?forum=vcmfcatl)

despite of those small points, do you like it ?

**Reply by smapelli** [reply]
Yes I do like it 👍

**Reply by pubpub-zz** [reply]
do not hesitate to star ppInk 😉

**Comment by smapelli** [comment]
Beware that playing around with the application I stumbled on a few other bugs. Using the move tool and with a few shapes drawn I got to a point where whatever shape I selected, the same one (the last drawn one) kept being moved. Another time playing with the lasso tool I don't know exactly how I landed on have a certain color pen selected and yet I could not draw anything. I landed on other bugs during my playing with the application and in general it seems to me there are numerous issues here and there still to be fixed and they appear easily if you just start trying out the different functionalities.

**Reply by smapelli** [reply]
Another suggestion is that the arrow tool should really draw the arrow starting from the initial point and not the final one. I believe from a conceptual point of view, as you have implemented it, you make it more complex to use it as a user needs to start away from the point he/she is really interested in highlighting or point to and then move toward it, Small things but from a user experience point of view they really make the difference.

**Reply by smapelli** [reply]
This other application I believe has a more immediate and accessible distribution of functionalities on tis palette, you might get further suggestions from it

https://www.goldgingko.com/screen-marker/index.htm

**Reply by pubpub-zz** [reply]
@smapelli,
* Can you please post your threads on ppInk discussions/Issues.
